document adjacent docs (sidecar) (#79056)
ansible/ansible: docs/docsite/rst/dev_guide/developing_locally.rst

============================================
If you are looking to add functionality to Ansible, you might wonder whether you need a module or a plugin. Here is a quick overview to help you understand what you need:

> * :ref:`Plugins <working_with_plugins>` extend Ansible's core functionality, most plugin types execute on the control node within the ``/usr/bin/ansible`` process. Plugins offer options and extensions for the core features of Ansible - transforming data, logging output, connecting to inventory, and more.

Reviewer: ```suggestion
* :ref:`Plugins <working_with_plugins>` extend Ansible's core functionality. Most plugin types execute on the control node within the ``/usr/bin/ansible`` process. Plugins offer options and extensions for the core features of Ansible: transforming data, logging output, connecting to inventory, and more.
```

> * Modules are a type of plugin that are meant for executions on the 'target' (usually that means on a remote system). They work as standalone scripts that Ansible executes in their own process outside of the controller. Modules interface with Ansible mostly via JSON, accepting arguments and returning information by printing a JSON string to stdout before exiting. Unlike the other plugins (which must be written in Python), modules can be written in any language; although Ansible projects only ship modules written in Python or Powershell.

Reviewer: ```suggestion
* Modules are a type of plugin that execute automation tasks on a 'target' (usually a remote system). Modules work as standalone scripts that Ansible executes in their own process outside of the controller. Modules interface with Ansible mostly via JSON, accepting arguments and returning information by printing a JSON string to stdout before exiting. Unlike the other plugins (which must be written in Python), modules can be written in any language; although Ansible provides modules in Python and Powershell only.
```

> * Modules are a type of plugin, the major difference is that they are meant for executions on the 'target' (usually that means on a remote system), they work as standalone scripts that Ansible executes in their own process outside of the controller. Modules interface with Ansible mosttly via JSON, accepting arguments and returning information by printing a JSON string to stdout before exiting. Unlike the other plugins (which must be written in Python) modules can be written in any language, though the Ansible projects only ships those written in Python or Powershell.

Reviewer: ```suggestion
* Modules are a type of plugin that are meant for executions on the 'target' (usually that means on a remote system). They work as standalone scripts that Ansible executes in their own process outside of the controller. Modules interface with Ansible mostly via JSON, accepting arguments and returning information by printing a JSON string to stdout before exiting. Unlike the other plugins (which must be written in Python), modules can be written in any language; although Ansible projects only ship modules written in Python or Powershell.
```

.. _local_modules:

Adding a module or plugin outside of a collection
==================================================

> You can configure Ansible to load standalone local modules or plugins in a specified location or locations and make them available to all playbooks and roles (using configured paths). Alternatively, you can make a non-collection local module or plugin available only to specific playbooks or roles (via adjacent paths).

Reviewer: ```suggestion
You can configure Ansible to load standalone local modules or plugins in specific locations and make them available to all playbooks and roles (using configured paths). Alternatively, you can make a non-collection local module or plugin available only to certain playbooks or roles (via adjacent paths).
```

* type ``ansible localhost -m my_local_module`` to see the output for that module, or
* type ``ansible-doc -t module my_local_module`` to see the documentation for that module

> .. note:: The same can be done for all plugin types, but using the specific configuration and/or adjacent directories for that plugin type, see below.

Reviewer: ```suggestion
.. note:: This applies to all plugin types but requires specific configuration and/or adjacent directories for each plugin type, see below.
```

.. note::

>    The ``ansible-doc`` command can parse module documentation from modules written in Python or an adjacent YAML file. If you have a module written in a programming language other than Python, please write the documentation in a Python or YAML file adjacent to the module file. :ref:`adjacent_yaml_doc`

Reviewer: ```suggestion
   The ``ansible-doc`` command can parse module documentation from modules written in Python or an adjacent YAML file. If you have a module written in a programming language other than Python, you should write the documentation in a Python or YAML file adjacent to the module file. :ref:`adjacent_yaml_doc`
```

* To use a standalone module only in a selected playbook or playbooks, store the module in a subdirectory called ``library`` in the directory that contains the playbook or playbooks.
* To use a standalone module only in a single role, store the module in a subdirectory called ``library`` within that role.

> .. note:: The same can be done for all plugin types, but using the specific directories for that plugin type, see below.

Reviewer: ```suggestion
.. note:: This applies to all plugin types but requires specific configuration and/or adjacent directories for each plugin type, see below.
```
